[TCL30-46] Show a button that allows the user to track device position changes

alxmcr opened the pull request:
## Description

As a user
I want to enable the geolocation
So that I can track the changes in my device (e.g. my smartphone)

## Related Issue

Closes [TCL30-46](https://the-collab-lab.atlassian.net/browse/TCL30-46)

## Acceptance Criteria

- Display the button: “Share Live Location“.
- If you have enabled your geolocation, you can share your location every time you change your position.
- If you clicked on the "Share Live Location" button, you should see new pins related to your new position.

## Type of Changes

<!-- Put an `✓` for the applicable box: -->

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/8689897/128402295-3b606c1e-1704-4da1-863e-bebb26a31a37.png)

### After

![image](https://user-images.githubusercontent.com/8689897/128423732-b3a61f61-5bdb-4534-8d5e-f9db83248177.png)

## Testing Steps / QA Criteria

1. Clone repository: `git clone https://github.com/the-collab-lab/tcl-30-whats-near-me`.
2. Change to PR branch: `git checkout jw-mm-ac-share-live-location-button`
3. Install NPM dependencies: `npm install`.
4. Run dev server: `npm run start`
5. Open your browser: `http://localhost:3000/` (in your device, eg. your smartphone)
6. Click on the "Share Live Location" button, and you should permit the Geolocation.
7. You walk or take a bus together with your device, and you can see new pins around you every time you change of position.

alxmcr committed:
Create component UserLocationPin

alxmcr committed:
Add style to UserLocationPin element

alxmcr committed:
Set default values to parameters in useLocations hook

alxmcr committed:
Add userCenterMap & userLocationShared as states in our MapCenter context

alxmcr committed:
Add a button called "Fly to me"

alxmcr committed:
Add UserLocationPin component to map

alxmcr committed:
Create FlyToMeButton component

alxmcr committed:
Extract the current logic to the FlyToMeButton component

alxmcr committed:
Merge branch 'main' into jw-mm-ac-fly-to-me-button

alxmcr committed:
Add a new state: trackingId

alxmcr committed:
Create a new button: ShareLiveLocationButton

alxmcr committed:
Add ShareLiveLocationButton component to navigation bar

alxmcr committed:
wip: add more information error message

laurenmbeatty commented:
I'm a little confused. I think I get the premise, though.

1. Share Live Location should allow sharing of geolocation and then put the map where you are.
2. If you drag the map somewhere, you should see pins around that new place, but you are not "there"
3. If I click Fly to Me...it should re-center on me, right?

Step 3 is not happening for me on this branch. On the `fly-to-me` PR, when I share my location, drag, and then click Fly to Me again, it goes back to me.

alxmcr commented:
> I'm a little confused. I think I get the premise, though.
> 
> 1. Share Live Location should allow sharing of geolocation and then put the map where you are.
> 2. If you drag the map somewhere, you should see pins around that new place, but you are not "there"
> 3. If I click Fly to Me...it should re-center on me, right?
> 
> Step 3 is not happening for me on this branch. On the `fly-to-me` PR, when I share my location, drag, and then click Fly to Me again, it goes back to me.

I'm so sorry for your confusion @laurenmbeatty .

1. Yes, **'Share Live Location button'** should do those things (share/change map center position also update your map center position 'automatically' if you **move or change your position with your device** - e.g. smartphone, and follow the user marker to you)
2. Yes, If you drag the map somewhere, the user marker shouldn't follow you, but you should get new pins around that new place.
3. Yes.

**Reason:** Fly to me button doesn't works

When I tested the first time this PR, I checked 'main' <- this branch (`jw-mm-ac-share-live-location-button`), but I changed '`jw-mm-ac-fly-to-me-button`' <- this branch. 

TIL: Second PR configuration about branches is not complete (only takes the commits your current branch` jw-mm-ac-share-live-location-button`)

I changed  `main `<- `jw-mm-ac-share-live-location-button`. You shouldn't have issues with the 'fly to me' button.


-------------------------------
Extra:

- **'Fly to me' button.** it's 'more visual' your behavior or difference between other values. First, a user marker is rendered (in your physical location). Second, if you drag the map, the user marker **doesn't move**, or **follow**' you, but you can see pins related to a current map center (if you dragged the map).
- **'Share Live Location.'** It's 'similar' to the 'fly to me' button because you start to share your location (in your current physical location) then the map center point should change to your current location, render a user marker, but it's different to 'fly to me' because the user marker should follow to you.
